libyul/optimiser/CommonSubexpressionEliminator.h

```diff
@@ -58,8 +58,21 @@ class CommonSubexpressionEliminator: public DataFlowAnalyzer
 	using ASTModifier::visit;
 	void visit(Expression& _e) override;
 
+	void assignValue(YulString _variable, Expression const* _value) override;
 private:
+	struct ExpressionHash {
+		uint64_t operator()(Expression const*) const;
+	};
+	struct ExpressionEqual {
+		bool operator()(Expression const*, Expression const*) const;
+	};
 	std::set<YulString> m_returnVariables;
+	std::unordered_map<
+		Expression const*,
```

Comment:
Can we directly use ``SyntacticallyEqual`` as comparison... functor :-)... if we switch to something like ``std::reference_wrapper<Expression const>`` here?